ikarus/finiteelements/fetraits.hh

```diff
@@ -80,6 +80,11 @@ struct FETraits
   using VectorType =
       std::conditional_t<useEigenRef, Eigen::Ref<Eigen::VectorX<ScalarType>>, Eigen::VectorX<ScalarType>&>;
 
+  /** \brief Type of the vector passed to updateState */
+  template <typename ScalarType = ctype>
+  using VectorTypeConst =
```

Comment:
Why not use `const VectorType`? in the correspodning locations?

Reply:
It was a workaround because we had troubles with the `const` qualifier along with `std::conditional_t<...>` (similar to https://godbolt.org/z/z78sGGWxv). But it works correctly here https://godbolt.org/z/hTcv7M5Ys somehow that we are trying to understand. I will look into it again and try to share the exact error message.

Reply:
I replicated the error here https://godbolt.org/z/qKrq1sPcq
The issue is that while using `const VectorType`, the `const` qualifier gets dropped. Hence we need `VectorTypeConst`. Is there a better way to fix this?

Reply:
There is always something new in C++.
If you have 

```cpp
using T= double&;
```
and

```cpp
using S= const T;
```
This does **not** get translated to
```cpp
using S= const double&;
```
This is due to the fact that the `const` does not applied only to the "double" part in `T` but also to the `&`.

So what you want is a reference to a constant `double` but your code is basically 

```cpp
using S= const (double&);
```

which means you want a constant reference to a double instead of a reference to a constant double.
But, references are always constant.
Similar if you have pointer `double*`

You can have 
```cpp
using S= double*; // pointer to double
using S= const double*; // pointer to constant double ( same as  double const *)
using S= const double* const; // constant pointer to constant double
using S= double* const; // constant pointer to double
```

The reference case `double&`  is similar to the last pointer case.
You have a reference to a double which you can change but you can't change the reference!
Therefore, references are always const and basically gets simplified to

```cpp
using S= double&;
```
since the `const` is redundant. Since it is similar to `double* const const`;

Therefore, this does not work an if the code is simplified further: https://godbolt.org/z/d4z16vTEY clang actually warns you that this const has no effect. `-Wignored-reference-qualifiers` 

To fix this finally you have the type `T` and the you strip the reference from it then add the const to the type and then again the reference.

That can be done with

```cpp
using S = std::remove_reference_t<T> const&;
```
see https://godbolt.org/z/Mr8qvKj4P .